src/Form/MetadataDisplayForm.php

```diff
@@ -502,11 +502,11 @@ public static function ajaxPreview($form, FormStateInterface $form_state) {
           $message = $exception->getMessage();
         }
       } finally {
-        if(!empty($message)) {
+        if (!empty($message)) {
           $preview_error = static::buildAjaxPreviewError($message);
           $output['preview_error'] = $preview_error;
         }
-        if (!$show_render_native || ($show_render_native && $mimetype != 'text/html')) {
+        if ($render && (!$show_render_native || ($show_render_native && $mimetype != 'text/html'))) {
```

Comment:
When `$render = $entity->renderNative($context);` (line 430 above) throws an exception, e.g. the template can't be rendered because of bad Twig, this generates a warning because `$render` is now undefined. Additionally, since the `$show_json_table` assignment comes after, it also generates the undefined warning in the logs. Should these variables be initialized before?

@DiegoPino thoughts?

Reply:
or maybe set $render to false or null in the catch if not set, or add an isset check for both variables?

Reply:
Ok, thought it through a little further, and it seems like it makes the most sense to use isset($render) instead of just $render in the conditionals. Will make another commit shortly.

Reply:
yep, that seems like easier. Thanks

Reply:
Or you can put another try/catch around $render = $entity->renderNative($context); ?

Reply:
> Or you can put another try/catch around $render = $entity->renderNative($context); ?

@DiegoPino Is there an advantage to nesting another try/catch? Seems like the outer one catches the same exception either way?

Reply:
Not really. I would require to refactor your try catch into two. The issue is because you have a final/ the catch itself (or the final) will have to do the extra work. Just dismiss. the isset() is  enough